src/providers/sagemaker.ts

```diff
@@ -22,42 +23,50 @@ import type { TransformContext } from '../util/transform';
 const sleep = (ms: number): Promise<void> => new Promise((resolve) => setTimeout(resolve, ms));
 
 /**
- * Options for configuring SageMaker provider
+ * Zod schema for validating SageMaker options
  */
-interface SageMakerOptions {
-  // AWS credentials options
-  accessKeyId?: string;
-  profile?: string;
-  region?: string;
-  secretAccessKey?: string;
-  sessionToken?: string;
-
-  // SageMaker specific options
-  endpoint?: string;
-  contentType?: string;
-  acceptType?: string;
-
-  // Model parameters
-  maxTokens?: number;
-  temperature?: number;
-  topP?: number;
-  stopSequences?: string[];
-
-  // Provider behavior options
-  delay?: number; // Delay between API calls in milliseconds
-  transform?: string; // Transform function or file path to transform prompts
-
-  // Model type for request/response handling
-  // TODO(Will): What is custom? User uploaded model?
-  // - Jumpstart is a model service, not a model type.
-  modelType?: 'openai' | 'llama' | 'huggingface' | 'jumpstart' | 'custom';
-
-  // Response format options
-  responseFormat?: {
-    type?: string;
-    path?: string; // JavaScript expression to extract content (formerly JSONPath)
-  };
-}
+export const SageMakerOptionsSchema = z
+  .object({
+    // AWS credentials options
+    accessKeyId: z.string().optional(),
+    profile: z.string().optional(),
+    region: z.string().optional(),
+    secretAccessKey: z.string().optional(),
+    sessionToken: z.string().optional(),
+
+    // SageMaker specific options
+    endpoint: z.string().optional(),
+    contentType: z.string().optional(),
+    acceptType: z.string().optional(),
+
+    // Model parameters
+    maxTokens: z.number().optional(),
```

Comment:
I am slightly reluctant to do this and think we should prefer to pass through instead when possible. We don't want promptfoo to break every time they add or remove a field from provider configuration in sagemaker. A pattern we follow in other places in the code base is to have validation on the important things. that we think will never change and then no validation on fields like maxTokens

Reply:
I largely agree except:

1. We're already hardcoding the request payload serializers which would need to be updated when they add or remove fields. One option to solve for both of these is moving the serialization logic onto the config, following the transform function pattern.
2. I've observed passing invalid values (e.g. incorrectly cased keys) resulting in opaque error messages; these errors should be caught as early in the callstack as possible and should be easy to fix.